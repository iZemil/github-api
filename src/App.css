.App {
  text-align: center;
}

.App-header {
  background-color: #222;
  padding: 20px;
  color: white;
}

.App-title {
  font-size: 1.5em;
}

.App-intro {
  font-size: large;
}

.container {
  width: 440px;
  margin: 0 auto;
  text-align: left;
}

.form-search-issues {
  width: 440px;
  margin: 20px auto;
  padding: 10px;
  border: 1px solid #000;
  border-radius: 3px;
  text-align: left;
}

input[type="text"] {
  width: 100%;
}

.table__row {
  width: 165px;
  margin-right: 10px;
  display: inline-block;
}

.table__row > label:not([class]) {
  width: 100%;
  display: inline-block;
  font-size: 13px;
}

.btn-submit {
  padding: 0 12px;
  cursor: pointer;
}

.issues {
  margin: 20px 0;
}

.issue {
  margin-bottom: 20px;
  padding: 10px;
  border: 1px solid #000;
}

.issue__title {
  font-weight: 600;
}

.issue__opened-by > span {
  display: inline-block;
  margin-right: 10px;
}