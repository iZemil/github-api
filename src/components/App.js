import React, { Component } from 'react';
import IssuesPage from '../containers/IssuesPage';
import Menu from './Menu';
import AboutPage from './AboutPage';
import IssueDetailPage from '../containers/IssueDetailPage';
import 'bootstrap/dist/css/bootstrap.css';
import {
  Route
} from 'react-router-dom';

class App extends Component {

  render() {
    return (
      <div className="App">
        <header className="App-header">
          <Menu />
        </header>
        <main className="container">
          <Route exact path="/" component={IssuesPage}/>
          <Route path="/about" component={AboutPage}/>
          <Route path="/detail/:id" component={IssueDetailPage}/>
        </main>
      </div>
    );
  }
}

export default App;
