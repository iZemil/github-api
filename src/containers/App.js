import React, { Component } from 'react';
import IssuesPage from './IssuesPage';
import Menu from '../components/Menu';
import AboutPage from '../components/AboutPage';
import IssueDetailPage from '../components/IssueDetailPage';
import 'bootstrap/dist/css/bootstrap.css';
import {
  Route
} from 'react-router-dom';

class App extends Component {

  render() {
    return (
      <div className="App">
        <header className="App-header">
          <Menu />
        </header>
        <main className="container">
          <Route exact path="/" component={IssuesPage}/>
          <Route path="/about" component={AboutPage}/>
          <Route path="/detail" component={IssueDetailPage}/>
        </main>
      </div>
    );
  }
}

export default App;
