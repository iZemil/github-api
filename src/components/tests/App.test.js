import React from 'react';
import Enzyme, {shallow} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import App from '../App';
Enzyme.configure({ adapter: new Adapter() });

test('Рендер компонента App без ошибок', () => {

  const app = shallow(
    <App />
  );
  expect(app).toMatchSnapshot();
});