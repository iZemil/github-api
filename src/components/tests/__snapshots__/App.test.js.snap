// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Рендер компонента App без ошибок 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <header
          className="App-header"
>
          <Example />
</header>,
        <main
          className="container"
>
          <Route
                    component={[Function]}
                    exact={true}
                    path="/"
          />
          <Route
                    component={[Function]}
                    path="/about"
          />
          <Route
                    component={[Function]}
                    path="/detail/:id"
          />
</main>,
      ],
      "className": "App",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <Example />,
          "className": "App-header",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": "header",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <Route
              component={[Function]}
              exact={true}
              path="/"
/>,
            <Route
              component={[Function]}
              path="/about"
/>,
            <Route
              component={[Function]}
              path="/detail/:id"
/>,
          ],
          "className": "container",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "component": [Function],
              "exact": true,
              "path": "/",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "component": [Function],
              "path": "/about",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "component": [Function],
              "path": "/detail/:id",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "main",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <header
            className="App-header"
>
            <Example />
</header>,
          <main
            className="container"
>
            <Route
                        component={[Function]}
                        exact={true}
                        path="/"
            />
            <Route
                        component={[Function]}
                        path="/about"
            />
            <Route
                        component={[Function]}
                        path="/detail/:id"
            />
</main>,
        ],
        "className": "App",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <Example />,
            "className": "App-header",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "header",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <Route
                component={[Function]}
                exact={true}
                path="/"
/>,
              <Route
                component={[Function]}
                path="/about"
/>,
              <Route
                component={[Function]}
                path="/detail/:id"
/>,
            ],
            "className": "container",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "component": [Function],
                "exact": true,
                "path": "/",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "component": [Function],
                "path": "/about",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "component": [Function],
                "path": "/detail/:id",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "main",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
